asdf/_tests/test_api.py

```diff
@@ -529,3 +529,21 @@ def test_none_values(tmp_path):
     with asdf.open(path) as af:
         assert "foo" in af
         assert af["foo"] is None
+
+
+def test_asdf_standard_version_tag_selection():
+    buff = io.BytesIO()
+
+    af = asdf.AsdfFile()
+    af.write_to(buff, version="1.0.0")
+    buff.seek(0)
+    content = buff.read()
+    assert b"!core/asdf-1.0.0" in content
+    assert b"!core/asdf-1.1.0" not in content
+
+    buff.seek(0)
+    af.write_to(buff, version="1.2.0")
```

Comment:
To check I understand this test, is it setting the ASDF standard to 1.0.0 then checking that core tag is version 1.0.0, as listed in the [1.0.0 manifest](https://github.com/asdf-format/asdf-standard/blob/main/src/asdf_standard/resources/manifests/asdf-format.org/core/core-1.0.0.yaml). Then setting the standard to 1.2.0 to check that the correct 1.1.0 core tag is used, as listed in the [1.2.0 manifest](https://github.com/asdf-format/asdf-standard/blob/main/src/asdf_standard/resources/manifests/asdf-format.org/core/core-1.2.0.yaml)?